src/plone/app/multilingual/browser/add.py

```diff
@@ -46,28 +41,63 @@ class AddViewTraverser(object):
     def __init__(self, context, request):
         self.context = context
         self.request = request
+        self.request.translation_info = dict()
+        self.info = self.request.translation_info
 
     def traverse(self, name, ignored):
-        ttool = getToolByName(self.context, 'portal_types')
-        ti = ttool.getTypeInfo(name)
-        if not IDexterityFTI.providedBy(ti):
-            # we are not on DX content
+        # Populate translation info
+        self.info['target_language'] = ILanguage(self.context).get_language()
+
+        catalog = getToolByName(self.context, 'portal_catalog')
+        # Search source object using unrestricted API,
+        # because user may be anonymous during traverse.
+        brains = catalog.unrestrictedSearchResults(UID=name)
+        if len(brains) != 1:
+            raise TraversalError(self.context, name)
+        source = brains[0]._unrestrictedGetObject()
```

Comment:
Naive me. It worked in live instance with restricted API, because I was translating public. I tried debugging this version and while traverser works for anonymous user, it seems that the returned add form has properly protected __call__ and.